src/audio/pipeline.c

```diff
@@ -389,11 +389,62 @@ static int pipeline_comp_hw_params(struct comp_dev *current,
 	return 0;
 }
 
+static int pipeline_comp_params_neg(struct comp_dev *current,
+				    struct comp_buffer *calling_buf,
+				    struct pipeline_walk_context *ctx,
+				    int dir)
+{
+	struct pipeline_data *ppl_data = ctx->comp_data;
+	uint32_t flags = 0;
+	int err = 0;
+
+	pipe_dbg(ppl_data->p, "pipeline_comp_params_neg(), current->comp.id = %u, dir = %u",
+		 dev_comp_id(current), dir);
+
+	/* return if current is already configured */
+	if (current->state != COMP_STATE_INIT &&
+	    current->state != COMP_STATE_READY) {
+		if (buffer_params_match(calling_buf,
```

Comment:
i.e. we are doingsomething here and not returning 0. inline comment here too.

Reply:
Thanks, will change it.